apps/mark-scan/backend/src/app.ts

```diff
@@ -101,6 +106,16 @@ function buildApi(
       return workspace.store.getSystemSettings() ?? DEFAULT_SYSTEM_SETTINGS;
     },
 
+    setPrecinctSelection(input: {
+      precinctSelection: PrecinctSelection | AllPrecinctsSelection;
+    }): void {
+      workspace.store.setPrecinctSelection(input.precinctSelection);
+    },
+
+    getPrecinctSelection(): Result<Optional<PrecinctSelection>, Error> {
+      return ok(workspace.store.getPrecinctSelection());
+    },
```

Comment:
Open to feedback on when to use `Result`. In this case, I use `Result` for `get` because precinct selection can be undefined but grout requires a defined response, so we meet that requirement by wrapping `undefined` in a `Result`. In `set` there's nothing to return so it felt unnecessary to return an empty `Result` 100% of the time.

Reply:
This reasoning makes sense to me!

Re the get method, the main alternative that I've seen is to return `null` when the store returns `undefined`, e.g.:

https://github.com/votingworks/vxsuite/blob/b26ecaf006712bf8caeec943e0bef68913e4c752/apps/central-scan/backend/src/central_scanner_app.ts#L209-L211

That might be preferable if we never expect to return an error `Result` (`err(...)`). But using a `Result` also seems totally fine.

Reply:
Grout actually doesn't require a defined response, it supports `undefined`. ReactQuery is the culprit here.

Given that, I personally think it makes the most sense to substitute in the `null` value in the `useQuery` hook to make it clearer why it's happening.

I think we're generally trying to use `Result` in API endpoints to signify errors that need to be shown to the user.